File: 1829-maximum-xor-for-each-query/1829-maximum-xor-for-each-query.cpp
class Solution {
public:
    vector<int> getMaximumXor(vector<int>& nums, int maximumBit) {
        int maxBit = (1 << maximumBit) - 1;
        vector<int> ans;
        vector<int> pxor(1, 0);

        // Build prefix XOR array
        for (int num : nums) {
            pxor.push_back(pxor.back() ^ num);
        }

        // Compute answers in reverse order
        for (int i = nums.size(); i >= 1; i--) {
            ans.push_back(pxor[i] ^ maxBit);
        }

        return ans;
    }
};

File: 1829-maximum-xor-for-each-query/1829-maximum-xor-for-each-query.py
class Solution:
    def getMaximumXor(self, nums: List[int], maximumBit: int) -> List[int]:
        max_bit = (2**maximumBit)-1
        ans = []
        pxor = [0]
        for i in nums:
            pxor.append(pxor[-1]^i)
        for i in range(len(nums), 0, -1):
            ans.append(pxor[i]^max_bit)            
        return ans